### PR TITLE
[Oxfordshire] add customer reference to old issues when sending comments

### PR DESCRIPTION
**Pull request by struan**
OCC issues need a customer_reference in extra to send but old issues
that were sent to Exor do not have this. If a problem is an old
issue then turn the external_id into the customer_reference.

Old issues are ones where the problem id and external_id do not match.

Fixes mysociety/fixmystreet-commercial#1214

Please check the following:

- [x] All cobrand-specific commits start their commit message with the cobrand in square brackets;
- [x] Is new functionality tested? CodeCov will warn you about the diff coverage, but won’t complain about e.g. new files;
- [x] Have you updated the changelog? If this is not necessary, put square brackets around this: [skip changelog]

Please check the contributing docs, and describe your pull request here.
Screenshots or GIF animations (using e.g. LICEcap) may be helpful.

Please include any issues that are fixed, using "fixes" or "closes" so that
they are auto-closed when the PR is merged.

Thanks for contributing!

### CODE REVIEW

File: perllib/FixMyStreet/Cobrand/Oxfordshire.pm [hunk 1]
```diff
@@ -179,6 +179,19 @@ sub open311_config_updates {
...
     my ($self, $update ) = @_;
 
+    my $p = $update->problem;
+
+    # Reports that were synced with Exor stored the external id
+    # in external_id so do not have a customer_reference. We need
+    # to copy the exor id across to customer_reference in order to
+    # send them
+    if (
+        $p->id != $p->external_id &&
```
[the remark this answers is not in the document]
Comment (struan) [reply perllib/FixMyStreet/Cobrand/Oxfordshire.pm]: Yes.

File: t/open311/post-service-request-updates.t [hunk 2]
```diff
@@ -75,6 +75,9 @@ my ($p3, $c3a) = p_and_c($bucks);
 my $c3b = c($p3, $other_user);
 my ($p4, $c4) = p_and_c($lewisham);
 
+# problem sent to WDM use FMS id as external_id
+$p2->update( { external_id => $p2->id } );
```
[the remark this answers is not in the document]
Comment (struan) [reply t/open311/post-service-request-updates.t]: Yup, but to a different value from the problem id so it looks as if it was sent to Exor. I've added a comment further down to hopefully explain this.

**Comment by LouiseMySociety** [comment]
If we haven't heard by the end of this sprint we will close.

**Comment by LouiseMySociety** [comment]
@struan to chase again this week.

**Comment by struan** [comment]
Closed as nothing back from customer

**Pull request closed without merging**